main.nf

```diff
@@ -293,6 +302,72 @@ if (!params.macs_gsize) {
              "  Please specify value for '--macs_gsize' to run these steps.\n" +
              "======================================================================="
 }
+///////////////////////////////////////////////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////
+/* --                                                                     -- */
+/* --                     FETCH READS FROM SRA                            -- */
+/* --                                                                     -- */
+///////////////////////////////////////////////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////
+
+
+if (params.accession_list) {
+
+    process GetSRA {
+        tag "${srr_id}"
+
+        input:
+        val(srr_id) from ch_srr_ids
+
+        output:
+        set val(srr_id), file("*.fastq.gz") into  sra_raw_reads
+
+        script:
+        """
+        fasterq-dump $srr_id --threads ${task.cpus} --split-3
+        pigz *.fastq
+        """
+    }
+
+    if ( params.s3_work && !params.single_end) {grouped_design_variables = sra_raw_reads.map { name, reads_tuple ->  tuple( name , [ name, 1 , "s3:/"+reads_tuple[0].toString(),  "s3:/"+reads_tuple[1].toString() ] ) }}
+    if (!params.s3_work && !params.single_end) {grouped_design_variables = sra_raw_reads.map { name, reads_tuple ->  tuple( name , [ name, 1 ,  reads_tuple[0], reads_tuple[1] ] ) } }
+
+    if ( params.s3_work && params.single_end) {grouped_design_variables = sra_raw_reads.map { name, reads ->  tuple( name , [ name, 1 , "s3:/"+reads.toString(),  ' ' ] ) }}
+    if (!params.s3_work && params.single_end) {grouped_design_variables = sra_raw_reads.map { name, reads ->  tuple( name , [ name, 1 ,  reads, ' ' ] ) } }
+
```

Comment:
Note here, to reconstruct the design for single end as described in the docs (see [here](https://github.com/TheJacksonLaboratory/atacseq/blob/s3-paths/assets/design_se.csv)) I am using a space in the single end and not empty string or null as those did work.
|||
|--|--|
 |`''` | ❌ |
 |`null` |  ❌|
 |`' '` | ✅ |